packages/backend/src/logging/caslAuditWrapper.ts

```diff
@@ -0,0 +1,201 @@
+import { type Ability } from '@casl/ability';
+import type { Rule as CaslRule } from '@casl/ability/dist/types/Rule';
+import { Abilities, type AnyObject } from '@casl/ability/dist/types/types';
+import { type SessionUser } from '@lightdash/common';
+import {
+    AuditActor,
+    AuditContext,
+    AuditResource,
+    AuditStatusType,
+    createAuditLogEvent,
+    type AuditLogEvent,
+} from './auditLog';
+
+export type AuditLogger = (event: AuditLogEvent) => void;
+
+export type AuditableUser = Pick<
+    SessionUser,
+    | 'userUuid'
+    | 'email'
+    | 'firstName'
+    | 'lastName'
+    | 'organizationUuid'
+    | 'role'
+>;
+
+type AuditHelperArgs = {
+    user: AuditableUser;
+    action: string;
+    subject: AnyObject;
+    ip?: string;
+    userAgent?: string;
+    requestId?: string;
+    ruleConditions?: string;
+};
+
+/**
+ * Creates an audit actor from a user
+ */
+const createActorFromUser = (user: AuditableUser): AuditActor => ({
+    id: user.userUuid,
+    email: user.email || '',
+    firstName: user.firstName || '',
+    lastName: user.lastName || '',
+    organizationUuid: user.organizationUuid || '',
+    organizationRole: user.role || 'unknown',
+    // TODO: Add group memberships
+    groupMemberships: [],
+});
+
+const createResourceFromSubject = (subject: AnyObject): AuditResource => {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { __caslSubjectType__, ...rest } = subject;
+
+    // Ensure we have a valid organizationUuid
+    const organizationUuid = (rest.organizationUuid as string) || '';
+
+    return {
+        type: (__caslSubjectType__ as string) || 'unknown',
+        id: (rest.id || rest.uuid || rest.projectUuid || organizationUuid) as
+            | string
+            | undefined,
+        name: (rest.name as string) || undefined,
+        organizationUuid,
+        projectUuid: (rest.projectUuid as string) || undefined,
+    };
+};
+
+const createContextFromArgs = (args: AuditHelperArgs): AuditContext => ({
+    organizationId: args.user.organizationUuid,
```

Comment:
`args.user.organizationUuid` or `args.subject.organizationUuid` ?

Reply:
The `actor` already has information about the user org id.
The `resource` already has information about the subject org&project id.
Should we remove project&org id from contex?

Reply:
Yeah I think the context is confusing here, I'm removing it!